File: Arrays/Kadanes.java
package Arrays;

public class Kadanes {
    public int maxSubArray(int[] nums) {
        int maxSum = Integer.MIN_VALUE, maxTillNow = 0;
        if (nums.length == 1) {
            return nums[0];
        }

        for (int i = 0; i < nums.length; i++) {
            maxTillNow += nums[i];

            if (maxTillNow > maxSum) {
                maxSum = maxTillNow;
            }
            if (maxTillNow < 0) {
                maxTillNow = 0;
            }
        }
        return maxSum;
    }
}

File: Kadanes.py
class Solution(object):
    def maxSubArray(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        if len(nums) == 1:
            return nums[0]
        maxSum=0
        maxTillNow=float('-inf')
        for i in nums:
            maxSum += i
            if maxSum > maxTillNow:
                maxTillNow = maxSum
            if maxSum <= 0:
                maxSum=0
        return maxTillNow
        